### PR TITLE
Bugfix: sets yesterday’s date when today is the first of the month

### PR DESCRIPTION
**Pull request by frankieroberto**
Previously when the day was the first of the month, clicking "Yesterday" set the date as the 0th of the month. This fix correctly sets it to the last day of the previous month, by using the [`setDate()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setDate) function on the date object, which if you pass `0`, sets the date object to the last day of the previous month (and previous year, if necessary).

See https://trello.com/c/hpFZ3R5K/170-yesterday-link-doesnt-work-on-1st-of-the-month

### CODE REVIEW

File: psd-web/app/assets/javascripts/date_input.js [hunk 1]
```diff
@@ -2,27 +2,31 @@ import $ from 'jquery';
 
 function dateInput(idPrefix) {
   $(document).ready(() => {
-    const dateObj = new Date();
-    const currentDay = dateObj.getUTCDate();
-    const currentMonth = dateObj.getUTCMonth() + 1;
-    const currentYear = dateObj.getUTCFullYear();
+    const currentDate = new Date();
+    var dateYesterday = new Date(currentDate.getTime())
```
Comment (StupidCodeFactory) [inline psd-web/app/assets/javascripts/date_input.js]: So according to this yesterday is the current day. Have I been leaving in the past all my life 😆 ??

Comment (frankieroberto) [reply psd-web/app/assets/javascripts/date_input.js]: 😂

**Comment by frankieroberto** [comment]
@StupidCodeFactory I think we should avoid creating new system tests – in cosmetics we managed to convert them all to feature specs using rack-test, which is way faster.

At some point I'd be happy to convert this to a proper component using progressive-enhancement, and at that point we could unit test the javascript using [jest](https://jestjs.io).

**Comment by edwardhorsford** [comment]
I'm not sure I can test this, short of waiting 2 days for the first of the month to roll by.

**Pull request merged**